src/reactionsystem.jl

```diff
@@ -1438,6 +1423,15 @@ function MT.flatten(rs::ReactionSystem; name = nameof(rs))
         metadata = MT.get_metadata(rs))
 end
 
+# Checks if a system is an allowed subsystem (i.e. no SDE parts and no jump).
+is_allowed_subsystem(sys::ReactionSystem) = true
+function is_allowed_subsystem(sys::System)
+    return (isnothing(MT.get_noise_eqs(sys)) || isempty(MT.get_noise_eqs(sys))) &&
+        (isnothing(MT.get_jumps(sys)) || isempty(MT.get_jumps(sys)))
```

Comment:
We should support non-reaction jumps now (and Brownians). They should be straightforward to just merge with the reaction jumps when converting later.

Reply:
(But probably makes sense to save those for follow-ups after you get the current functionality working here…)

Reply:
Yeah, I was going to give this one (and some other composability stuff) to you to re-do as you see best, right now I just try to make as much as possible pass as soon as possible.